### DISCUSSION TITLE
how to manage the filter?

### DISCUSSION THREAD

**Discussion by tiretrak** [marcel-licence/esp32_basic_synth#22] [Q&A]
how to change cutoff on midi ?

**Comment by marcel-licence** [comment]
I've made an update.
In z_config.into you will find a mapping with Synth_SetParam. Here is the associated channel and cc mapped to different parameters. Does this help?

Current mapping:
 { 0x5, 0x10, "R6", NULL, Synth_SetParam, SYNTH_PARAM_MAIN_FILT_CUTOFF},

My MIDI controller has a bit weird mapping so with default settings you can change the cutoff by cc: 0x10 (16) on ch: 5 (6t ch)

**Reply by tiretrak** [reply]
I cannot verify.
the new version does not compile
F:\!!!Arduino\esp32_basic_synth-main19052020\esp32_basic_synth\adc_module.ino: In function 'void AdcMul_Init()':
adc_module:56:22: error: 'analogSetCycles' was not declared in this scope
     analogSetCycles(1);
                      ^
F:\!!!Arduino\esp32_basic_synth-main19052020\esp32_basic_synth\adc_module.ino: In function 'void AdcMul_Process()':
adc_module:96:20: error: 'adcStart' was not declared in this scope
         adcStart(12);
                    ^
adc_module:100:27: error: 'adcBusy' was not declared in this scope
             if (adcBusy(12) == false)
                           ^
adc_module:102:38: error: 'adcEnd' was not declared in this scope
                 readAccu += adcEnd(12);
                                      ^
adc_module:106:18: error: 'adcEnd' was not declared in this scope
         adcEnd(12);

**Reply by marcel-licence** [reply]
I try to figure out where the functions come from. It takes some time

**Reply by marcel-licence** [reply]
Using board 'esp32doit-devkit-v1' from platform in folder: C:\Users\<USERNAME>\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.4
Using core 'esp32' from platform in folder: C:\Users\<USERNAME>\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.4

Please check if you using the correct board. If the version is different please let me know.

**Reply by tiretrak** [reply]
i have \esp32\1.0.6
change to 1.0.4 and the program compiled successfully

**Reply by marcel-licence** [reply]
Thanks for your feedback. I am just wondering why they removed the functions in the latest library.
